documentation/SolutionsRegistryFormat.md

```diff
@@ -143,6 +143,37 @@ This directive is used to detect whether a solution is installed. If any of thes
 ]
 ```
 
+####isConfigurable
+This is run before configuration to ensure that the application is actually ready to be configured. This is relevant for applications where e.g. a configuration file needs to be present, a wizard needs to be run on the first launch, etc. Each of the blocks will be evaluated, if *any* of them evaluates to true, the solution is considered configurable. The absence of an `isConfigurable` block is also interpreted as the solution being configurable.
+
+If a solution is not configurable, the `makeConfigurable` block will be executed immediately following the `isConfigurable` check (see below).
+
+**Example Entry**:
+```
+"isConfigurable": [
+    {
+        "type": "gpii.lifecycleActions.pathExists",
+        "path": "/tmp/fakemag1.settings.json"
+    }
+]
+```
+
+####makeConfigurable
+Is the actions that need to be taken to make the application configurable (such as running a wizard, creating a default configuration file, adding a new system user, etc). This will be triggered by the system if `isConfigurable` has evaluated to false and run immediately following the `isConfigurable` check.
+
+**Example Entry**:
+```
+"makeConfigurable": [
+    {
+        "type": "gpii.lifecycleActions.createFile",
```

Comment:
no - I completely agree, that if there is no proper use case it should be mothballed, else it'll just pollute the code base, risk going stale, etc..

 I talked with @javihernandez yesterday, and it does indeed seem that ORCA (and, I believe NVDA) needs extra actions the first time they're used. Though the use case there is that they need to be started and then stopped (which will trigger their initial setup routine), before the settings are set. Anyway, I'll let him confirm/unconfirm this